packages/web5/lib/src/pexv2/pd.dart

```diff
@@ -0,0 +1,274 @@
+import 'dart:math';
+import 'dart:typed_data';
+import 'package:convert/convert.dart';
+import 'package:json_path/json_path.dart';
+import 'package:json_schema/json_schema.dart';
+
+/// PresentationDefinition represents a DIF Presentation Definition defined [here].
+/// Presentation Definitions are objects that articulate what proofs a Verifier requires.
+///
+/// [here]: https://identity.foundation/presentation-exchange/#presentation-definition
+class PresentationDefinition {
+  String id;
+  String? name;
+  String? purpose;
+  List<InputDescriptor> inputDescriptors;
+
+  PresentationDefinition({
+    required this.id,
+    this.name,
+    this.purpose,
+    required this.inputDescriptors,
+  });
+
+  factory PresentationDefinition.fromJson(Map<String, dynamic> json) =>
+      PresentationDefinition(
+        id: json['id'],
+        name: json['name'],
+        purpose: json['purpose'],
+        inputDescriptors: List<InputDescriptor>.from(
+          json['input_descriptors'].map((x) => InputDescriptor.fromJson(x)),
+        ),
+      );
+
+  Map<String, dynamic> toJson() => {
+        'id': id,
+        'name': name,
+        'purpose': purpose,
+        'input_descriptors':
+            List<dynamic>.from(inputDescriptors.map((x) => x.toJson())),
+      };
+
+  List<String> selectCredentials(List<String> vcJwts) {
+    final Set<String> matches = {};
+
+    for (final inputDescriptor in inputDescriptors) {
+      final matchingVcJwts = inputDescriptor.selectCredentials(vcJwts);
+      if (matchingVcJwts.isEmpty) {
+        return [];
+      }
+      matches.addAll(matchingVcJwts);
+    }
+
+    return matches.toList();
+  }
+}
+
+class _TokenizedField {
+  List<String> paths;
+  String token;
+
+  _TokenizedField({required this.paths, required this.token});
+}
+
+/// InputDescriptor represents a DIF Input Descriptor defined [here].
+/// Input Descriptors are used to describe the information a Verifier requires of a Holder.
+///
+/// [here]: https://identity.foundation/presentation-exchange/#input-descriptor
+class InputDescriptor {
+  String id;
+  String? name;
+  String? purpose;
+  Constraints constraints;
+
+  InputDescriptor({
+    required this.id,
+    this.name,
+    this.purpose,
+    required this.constraints,
+  });
+
+  factory InputDescriptor.fromJson(Map<String, dynamic> json) =>
+      InputDescriptor(
+        id: json['id'],
+        name: json['name'],
+        purpose: json['purpose'],
+        constraints: Constraints.fromJson(json['constraints']),
+      );
+
+  Map<String, dynamic> toJson() => {
+        'id': id,
+        'name': name,
+        'purpose': purpose,
+        'constraints': constraints.toJson(),
+      };
+
+  String _generateRandomToken() {
+    final rand = Random.secure();
+    final bytes = Uint8List(16);
+    for (int i = 0; i < 16; i++) {
+      bytes[i] = rand.nextInt(256);
+    }
+    return hex.encode(bytes);
+  }
+
+  List<String> selectCredentials(List<String> vcJWTs) {
```

Comment:
`vcJwts` to follow pre-existing convention in this repo

Reply:
fixed